Python/pythonrun.c

```diff
@@ -564,6 +564,30 @@ PyRun_SimpleStringFlags(const char *command, PyCompilerFlags *flags)
     return _PyRun_SimpleStringFlagsWithName(command, NULL, flags);
 }
 
+static int
+parse_exit_code(PyObject *code, int *exitcode_p)
+{
+    if (PyLong_Check(code)) {
+        // gh-125842: Use a long long to avoid an overflow error when `long`
+        // is 32-bit. We still truncate the result to an int.
+        int exitcode = (int)PyLong_AsLongLong(code);
+        if (exitcode == -1 && PyErr_Occurred()) {
+            // On overflow or other error, clear the exception and use -1
+            // as the exit code to match historical Python behavior.
+            PyErr_Clear();
```

Comment:
I'm actually wondering but should the docs be amended for `sys.exit`:

> If an error occurs in the cleanup after the Python interpreter has caught [SystemExit](https://docs.python.org/3.12/library/exceptions.html#SystemExit) (such as an error flushing buffered data in the standard streams), the exit status is changed to 120

Or is `parse_exit_code` not part of the "cleanup" part?

Reply:
I'm not sure it's worth documenting -- explaining this idiosyncratic behavior seems more likely to be confusing than helpful.

I think for 3.14 we can consider trying to make it consistent with `_Py_Finalize()`, where errors cause the exit status to be to 120, but I'd like to keep the behavioral change in the bugfix to be as small as possible

Reply:
Fine by me, I just wanted to be sure on this matter.